doc/getting-started/gsg_quick_setup.sh

```diff
@@ -0,0 +1,277 @@
+    echo "Getting latest Service OS kernel version: $new_kernel"
+    cur_kernel=`cat /mnt/loader/entries/acrn.conf | sed -n 2p | cut -d'/' -f4`
+    echo "Getting current Service OS kernel version: $cur_kernel"
+
+    echo "Replacing root partition uuid in acrn.conf"
+    sed -i "s/<UUID of rootfs partition>/`blkid -s PARTUUID -o value $root_partition`/g" /mnt/loader/entries/acrn.conf
+
+    if [[ "$cur_kernel" == "$new_kernel" ]]; then
+        echo "No need to replace kernel conf info"
+    else
+        echo "Replace with new sos kernel in acrn.conf"
+        sed -i "s/$cur_kernel/$new_kernel/g" /mnt/loader/entries/acrn.conf
+    fi
+
+    echo "Service OS setup done!"
+    echo "Rebooting Service OS to taking effect changes."
+else
+    echo -e "Fail to upgrade sos $target_clearlinux from mirror url."
+fi
+
+umount /mnt
+sync
+reboot
+}
+
+function upgrade_uos()
+{
+# UOS download link
+uos_image_link="https://download.clearlinux.org/releases/$target_version/clear/clear-$target_version-kvm.img.xz"
+
+# Set proxy if needed.
+if [[ -n $proxy ]]; then
+    echo "Setting proxy to: $proxy"
+    export https_proxy=$proxy
+fi
+
+if [[ ! -z `findmnt /mnt -n` ]]; then umount /mnt; fi
+
+# Do upgrade uos process.
+if [[ $skip_download_uos == 1 ]]; then
+    uos_img_xz=$(find ~/ -name clear-$target_version-kvm.img.xz)
+    uos_img=$(find ~/ -name clear-$target_version-kvm.img)
+    if [[ ! -f $uos_img_xz ]] && [[ ! -f $uos_img ]]; then
+        echo "You should download uos clear-$target_version-kvm.img.xz file firstly." && exit 1
+    fi
+    if [[ -f $uos_img_xz ]]; then
+        echo "Unxz uos file: $uos_img_xz"
+        unxz $uos_img_xz
+        uos_img=`echo $uos_img_xz | sed 's/.xz$//g'`
+    fi
+    echo "get uos img file: $uos_img"
+    if [[ $? -eq 0 ]]; then
+        uos_partition=`losetup -f -P --show $uos_img`p3
+    else
+        echo "fail to losetup uos img, please check uos img status" && exit 1
+    fi
+    mount $uos_partition /mnt
+    cp -r /usr/lib/modules/"`readlink /usr/lib/kernel/default-iot-lts2018 | awk -F '2018.' '{print $2}'`.iot-lts2018" /mnt/lib/modules
+else
+    cd ~
+    echo "Downloading UOS image: $uos_image_link"
+    curl $uos_image_link -o clear-$target_version-kvm.img.xz
+    uos_img=clear-$target_version-kvm.img
+    if [[ -f $uos_img ]] && [[ -f $uos_img.xz ]]; then echo "Moving $uos_img to $uos_img.old."; mv $uos_img $uos_img.old; fi
+    if [[ $? -eq 0 ]]; then
+        echo "Unxz UOS image: clear-$target_version-kvm.img.xz"
+        unxz clear-$target_version-kvm.img.xz
+    fi
+    if [[ $? -eq 0 ]]; then
+        uos_partition=`losetup -f -P --show $uos_img`p3
+    else
+        echo "Missing UOS file: $uos_img, please check if UOS is exist." && exit 1
+    fi
+    mount $uos_partition /mnt
+    cp -r /usr/lib/modules/"`readlink /usr/lib/kernel/default-iot-lts2018 | awk -F '2018.' '{print $2}'`.iot-lts2018" /mnt/lib/modules
+fi
+umount /mnt
+losetup -D
```

Comment:
The user may be using other loop devices for other things, we shouldn't detach *all* of them unconditionally.

Reply:
Removed this operation